gdb/mips-tdep.c

```diff
@@ -7684,7 +7692,7 @@ micromips_insn_at_pc_has_delay_slot (struct gdbarch *gdbarch,
 	return 0;
     }
 
-  return micromips_instruction_has_delay_slot (insn, mustbe32);
+  return micromips_instruction_has_delay_slot (gdbarch, insn, mustbe32);
```

Comment:
I think we can leave the micromips changes alone as we don't support compressed MIPS instructions on CHERI AFAIK.  CHERI binaries will always use the mips32_* routines.  I think the assert in micromips_next_pc() is fine, but adding the gdbarch arg here makes the diff a bit larger which might result in more conflicts in future merges.